### ISSUE TITLE
Expose user home directories to dask workers

### ISSUE THREAD

**Issue by yuvipanda** [pangeo-data/pangeo#88]
(I think this idea came from the UK Met office folks! Tag yourself ? :D)

We'd like to mount user home directories on all the dask workers. Unfortunately we now provision a Google Persistent Disk for each user, and those can be attached to only one pod at a time.

A reasonable way around this seems to be:

1. Run a [sidecar](http://blog.kubernetes.io/2015/06/the-distributed-system-toolkit-patterns.html) container on each user notebook that's running [user space NFS](https://github.com/nfs-ganesha/nfs-ganesha/wiki). Kernel mode NFS is also a possibility, but I know that a lot of new distros are built without NFS support built in to the kernel, so user mode might be more widely applicable (and needs less permissions!)
2. Sidecar should have access to user's home directory (readonly?), and export that
3. Dask workers can just mount the exported homedir by directly accessing the notebook Pod's IP (which we know). This can just use the default NFS volume provider from Kubernetes, although at high mount counts you might want something like a better documented version of http://github.com/yuvipanda/nfs-flex-volume. This also needs vetting on which base OS images support this.

There are some security considerations here, since we'd need to restrict which pods can mount the homedirs of which users. I only know of IP whitelisting for NFS export security, which might be slightly racy to implement in our case (since we'll need to know target IP of dask worker *before* it is fully started), but there are probably other NFS security methods I'm unaware of.

**Comment by mrocklin** [comment]
cc @jacobtomlinson and @niallrobinson

**Comment by jacobtomlinson** [comment]
Yes this would be useful to us as we use [EBS](https://aws.amazon.com/ebs/) volumes for user home directories which are single mount only.

Our long term plan was to switch the home directories to NFS once [EFS](https://aws.amazon.com/efs/) is available in the London AWS region. This way the workers could access them in the same way the notebook does.

However I do agree a sidecar NFS server provides the most flexibility and doesn't make assumptions about how home directories are mounted. I'll start looking at this today and tomorrow.
